packages/hash/api/src/graphql/typeDefs/page.typedef.ts

```diff
@@ -269,11 +308,26 @@ export const pageTypedef = gql`
   permit unions as input to a mutation
   """
   input UpdatePageAction {
-    insertNewBlock: InsertNewBlock
+    insertBlock: InsertBlock
     removeBlock: RemoveBlock
     moveBlock: MoveBlock
     updateEntity: UpdateEntity
     swapBlockData: SwapBlockData
+    createEntity: CreateEntityAction
+    createEntityType: CreateEntityTypeAction
```

Comment:
I agree with the sentiment, but this is still page-specific so I'd be minded to keep 'Page' in there somewhere

Reply:
> but this is still page-specific so I'd be minded to keep 'Page' in there somewhere

@CiaranMn it's my understanding that eventually all changes to entities will go through collab, whether it is to entities that are in pages or not (for example updates to user entities). Is that incorrect?

Reply:
That's correct, but this mutation remains Page-specific, since a big part of what it is doing inserting, removing and moving blocks around a Page. The return type specifies a `page`, and it will throw an error if the provided entity is not a page (since it is finding it through the Page model).

It could be rewritten to be more generic, but I think we would be better off having a smaller, more generic mutation for doing things that don't involve all the block-specific logic in this mutation.

Reply:
> It could be rewritten to be more generic, but I think we would be better off having a smaller, more generic mutation for doing things that don't involve all the block-specific logic in this mutation.

I guess I already consider the method to be more generic than its name implies because to my understanding it now can create an entity or entity type (whether or not they are used in the page). In those use-cases I don't think having `Page` as a return-type is very intuitive/logical either, shouldn't the return type to be the responses of the actions (whether that's a `Page`, a new `Entity`, a new `Entity Type`, etc?

@nathggns correct me if I'm wrong, but my instinct is having distinct `updatePageContents` and `processAtomicActions` mutations would prevent actions that are accepted by one and not the other from being run in the same transaction atomically, which seems like it partially defeats the use-case of the methods in some use cases.

Reply:
Made a task to track this, but I think it's beyond the scope of this PR https://app.asana.com/0/1201095311341924/1202417508635353/f